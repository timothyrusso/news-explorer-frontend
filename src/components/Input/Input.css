.popup__input {
  display: block;
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  border: none;
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  max-width: 358px;
  width: 100%;
  margin: 0 auto 29.74px;
  padding: 0 0 9px;
  color: #000;
  opacity: 0.2;
}

@media screen and (max-width: 552px) {
  .popup__input {
    max-width: 238px;
    margin-bottom: 30px;
  }
}

.popup__input_type_error {
  color: rgba(255, 0, 0, 1);
  margin: 0;
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 1;
  width: 100%;
  max-width: 358px;
  display: block;
  position: relative;
  left: 36px;
  bottom: 25px;
}

@media screen and (max-width: 552px) {
  .popup__input_type_error {
    left: 22px;
  }
}

.popup__input-label {
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 15px;
  color: #2F71E5;
  margin: 0 0 9px 36px;
  display: block;
}

input:focus {
  color: #000;
  opacity: 1;
}
