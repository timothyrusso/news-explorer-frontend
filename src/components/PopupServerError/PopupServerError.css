.popup-server-error {
  font-family: 'Inter', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 1;
  margin: 0;
  color: #FF0000;
  display: flex;
  justify-content: center;
}
