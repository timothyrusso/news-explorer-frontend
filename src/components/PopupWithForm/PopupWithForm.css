.popup__title {
  font-family: 'Roboto', Arial, sans-serif;
  font-style: normal;
  font-weight: 900;
  font-size: 24px;
  line-height: 28px;
  color: #000;
  max-width: 358px;
  width: 100%;
  margin: 0 auto;
  padding: 34px 0 22px;
}

@media screen and (max-width: 552px) {
  .popup__title {
    max-width: 238px;
    font-size: 18px;
    padding: 25px 0 75px;
  }
}

.submit-button {
  display: block;
  width: 100%;
  max-width: 358px;
  min-height: 64px;
  border: none;
  background-color: #2F71E5;
  color: #fff;
  margin: 39px auto 0;
  border-radius: 100px;
  font-family: 'Roboto', Arial, sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 18px;
  line-height: 24px;
  padding: 0;
  transition: opacity 0.2s linear;
}

.submit-button:hover {
  opacity: 0.8;
  cursor: pointer;
}

@media screen and (max-width: 552px) {
  .submit-button {
    max-width: 238px;
    font-size: 14px;
    margin: 45px auto 25px;
  }
}

.submit-button_disabled {
  background-color: rgba(0, 0, 0, 0);
  color: rgba(0, 0, 0, 1);
  opacity: 0.2;
  border: 1px solid rgba(0, 0, 0, 1);
  cursor: default;
  pointer-events: none;
}
