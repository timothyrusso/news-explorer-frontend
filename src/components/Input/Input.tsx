import React, { FC } from 'react';
import { ErrorMessage } from '../../store/errors/error.type';
import './Input.css';

type InputProps = {
  type: string;
  idName: string;
  name: string;
  fieldName: string;
  placeholder: string;
  onChange: (evt: React.ChangeEvent<HTMLInputElement>) => void;
  minLength?: number;
  maxLength?: number;
  value: string;
  errorMessage: ErrorMessage | {};
  labelText: string;
};

const Input: FC<InputProps> = ({
  type,
  idName,
  name,
  fieldName,
  placeholder,
  onChange,
  minLength,
  maxLength,
  value,
  errorMessage,
  labelText,
}) => {
  return (
    <>
      <label htmlFor={idName} className="popup__input-label">
        {labelText}
      </label>
      <input
        type={type}
        id={idName}
        name={name}
        className={`popup__input popup__input_${fieldName}`}
        placeholder={placeholder}
        required
        minLength={minLength}
        maxLength={maxLength}
        value={value}
        onChange={onChange}
      />
      <span id={`${idName}-error`} className="popup__input_type_error">
        {(errorMessage as ErrorMessage)[name]}
      </span>
    </>
  );
};

export default Input;
