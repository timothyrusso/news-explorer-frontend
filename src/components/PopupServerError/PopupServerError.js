import "./PopupServerError.css";

const PopupServerError = ({ popupServerError }) => {
  return <p className="popup-server-error">{popupServerError}</p>;
};

export default PopupServerError;
