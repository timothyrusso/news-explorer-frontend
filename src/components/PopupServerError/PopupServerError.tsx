import React, { FC } from 'react';
import './PopupServerError.css';

type PopupServerErrorProps = {
  popupServerErrorMessage: string;
};

const PopupServerError: FC<PopupServerErrorProps> = ({
  popupServerErrorMessage,
}) => {
  return <p className="popup-server-error">{popupServerErrorMessage}</p>;
};

export default PopupServerError;
