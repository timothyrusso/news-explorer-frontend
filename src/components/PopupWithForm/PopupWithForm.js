import Popup from "../Popup/Popup";
import Form from "../Form/Form";
import "./PopupWithForm.css";
import FormRedirect from "../FormRedirect/FormRedirect";

const PopupWithForm = ({
  name,
  title,
  isOpen,
  onClose,
  onSwitch,
  popupRedirectText,
  buttonText,
  onSubmit,
  loadingText,
  isLoadingText,
  formValidity,
  onFormUpdate,
  children,
}) => {
  return (
    <Popup isOpen={isOpen} name={name} onClose={onClose}>
      <Form
        name={`myForm${name}`}
        onSubmit={onSubmit}
        onFormUpdate={onFormUpdate}
      >
        <h2 className="popup__title">{title}</h2>
        {children}
        <button
          type="submit"
          className={`submit-button popup__button ${
            !formValidity ? "submit-button_disabled" : ""
          }`}
          disabled={!formValidity}
        >
          {isLoadingText ? loadingText : buttonText}
        </button>
        <FormRedirect
          onSwitch={onSwitch}
          popupRedirectText={popupRedirectText}
        />
      </Form>
    </Popup>
  );
};

export default PopupWithForm;
