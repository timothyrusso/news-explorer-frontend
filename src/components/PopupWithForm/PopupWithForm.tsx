import React, { FC, ReactNode } from 'react';
import Popup from '../Popup/Popup';
import Form from '../Form/Form';
import FormRedirect from '../FormRedirect/FormRedirect';
import { useAppSelector } from '../../hooks/useAppSelector';
import './PopupWithForm.css';

type PopupWithFormProps = {
  name: string;
  title: string;
  isOpen: boolean;
  popupRedirectText: string;
  buttonText: string;
  onSubmit: (evt: React.FormEvent<HTMLFormElement>) => void;
  loadingText: string;
  isLoadingText: boolean;
  children: ReactNode;
};

const PopupWithForm: FC<PopupWithFormProps> = ({
  name,
  title,
  isOpen,
  popupRedirectText,
  buttonText,
  onSubmit,
  loadingText,
  isLoadingText,
  children,
}) => {
  const formValidity = useAppSelector((state) => state.errors.formValidity);
  const infoTooltip = useAppSelector(
    (state) => state.toggles.isInfoTooltipOpen
  );

  return (
    <Popup isOpen={isOpen} name={name} infoTooltip={infoTooltip}>
      <Form name={`myForm${name}`} onSubmit={onSubmit}>
        <h2 className="popup__title">{title}</h2>
        {children}
        <button
          type="submit"
          className={`submit-button popup__button ${
            !formValidity ? 'submit-button_disabled' : ''
          }`}
          disabled={!formValidity}
        >
          {isLoadingText ? loadingText : buttonText}
        </button>
        <FormRedirect popupRedirectText={popupRedirectText} />
      </Form>
    </Popup>
  );
};

export default PopupWithForm;
