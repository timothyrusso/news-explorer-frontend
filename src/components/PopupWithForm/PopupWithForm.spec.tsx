import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store/store';
import PopupWithForm from './PopupWithForm';
import Input from '../Input/Input';
import PopupServerError from '../PopupServerError/PopupServerError';

const MIN_LENGTH = 5;
const MAX_LENGTH = 20;

const mockChildren = (
  <>
    <Input
      type="email"
      idName="email-input"
      name="email"
      fieldName="field_email"
      placeholder="Enter email"
      minLength={MIN_LENGTH}
      maxLength={MAX_LENGTH}
      value=""
      onChange={jest.fn()}
      errorMessage=""
      labelText="Email"
    />
    <Input
      type="password"
      idName="password-input"
      name="password"
      fieldName="field_password"
      placeholder="Enter password"
      minLength={MIN_LENGTH}
      maxLength={MAX_LENGTH}
      value=""
      onChange={jest.fn()}
      errorMessage=""
      labelText="Password"
    />
    <Input
      type="text"
      idName="username-input"
      name="username"
      fieldName="field_username"
      placeholder="Enter your username"
      minLength={MIN_LENGTH}
      maxLength={MAX_LENGTH}
      value=""
      onChange={jest.fn()}
      errorMessage=""
      labelText="Username"
    />
    <PopupServerError popupServerErrorMessage="" />
  </>
);

describe('PopupWithForm component', () => {
  it('Should render the PopupWithForm component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <PopupWithForm
            name="signup"
            title="Sign up"
            isOpen={true}
            popupRedirectText="popupRedirectText"
            buttonText={'Sign up'}
            loadingText={'Saving..'}
            isLoadingText={false}
            onSubmit={jest.fn()}
          >
            {mockChildren}
          </PopupWithForm>
        </Router>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
