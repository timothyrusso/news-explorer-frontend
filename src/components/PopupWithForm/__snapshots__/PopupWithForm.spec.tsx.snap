// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupWithForm component Should render the PopupWithForm component 1`] = `
<DocumentFragment>
  <div
    class="popup popup_opened popup_type_signup"
  >
    <div
      class="popup__container"
    >
      <form
        action="#"
        class="popup__form"
        name="myFormmyFormsignup"
      >
        <h2
          class="popup__title"
        >
          Sign up
        </h2>
        <label
          class="popup__input-label"
          for="email-input"
        >
          Email
        </label>
        <input
          class="popup__input popup__input_field_email"
          id="email-input"
          maxlength="20"
          minlength="5"
          name="email"
          placeholder="Enter email"
          required=""
          type="email"
          value=""
        />
        <span
          class="popup__input_type_error"
          id="email-input-error"
        />
        <label
          class="popup__input-label"
          for="password-input"
        >
          Password
        </label>
        <input
          class="popup__input popup__input_field_password"
          id="password-input"
          maxlength="20"
          minlength="5"
          name="password"
          placeholder="Enter password"
          required=""
          type="password"
          value=""
        />
        <span
          class="popup__input_type_error"
          id="password-input-error"
        />
        <label
          class="popup__input-label"
          for="username-input"
        >
          Username
        </label>
        <input
          class="popup__input popup__input_field_username"
          id="username-input"
          maxlength="20"
          minlength="5"
          name="username"
          placeholder="Enter your username"
          required=""
          type="text"
          value=""
        />
        <span
          class="popup__input_type_error"
          id="username-input-error"
        />
        <p
          aria-live="polite"
          class="popup-server-error"
        />
        <button
          aria-label="Submit Sign up"
          class="submit-button popup__button  "
          type="submit"
        >
          Sign up
        </button>
        <p
          class="form-redirect"
        >
          or 
          <button
            aria-label="test"
            class="form-redirect_type_link"
            type="button"
          >
            popupRedirectText
          </button>
        </p>
      </form>
      <button
        aria-label="Close signup popup"
        class="close-button "
        type="button"
      />
    </div>
  </div>
</DocumentFragment>
`;
